lib/saluki-components/src/sources/checks/builder/python/builder.rs

```diff
@@ -54,19 +58,32 @@ impl Check for PythonCheck {
 static PYTHON_ENV_BUILDER: OnceLock<PythonEnvBuilder> = OnceLock::new();
 
 struct PythonEnvBuilder {
+    check_events_tx: Sender<CheckMetric>,
     agent_check_class: Option<PyObject>,
+    custom_checks_folders: Option<Vec<String>>,
 }
 impl PythonEnvBuilder {
-    fn get_instance(custom_checks_folders: &Option<Vec<String>>) -> &'static PythonEnvBuilder {
-        PYTHON_ENV_BUILDER.get_or_init(|| PythonEnvBuilder::initialize(custom_checks_folders))
+    pub fn new(check_events_tx: Sender<CheckMetric>, custom_checks_folders: Option<Vec<String>>) -> Self {
+        Self {
+            check_events_tx,
+            custom_checks_folders,
+            agent_check_class: None,
+        }
+    }
+
+    fn get_instance(&self) -> &'static PythonEnvBuilder {
+        PYTHON_ENV_BUILDER.get_or_init(|| self.initialize())
     }
```

Comment:
This is a little weird, having `PythonEnvBuilder::new` and `PythonEnvBuilder::initialize` both return `Self`.

I would just do something like:

- change `PYTHON_ENV_BUILDER` to `PYTHON_ENV_INITIALIZE: OnceLock<()>`
- change `PythonEnvBuilder::new` to do all of the initialization logic and return `Result<Self, GenericError>`
- in `PythonEnvBuilder::new`, have it try to claim the right to initialize the Python interpreter by calling `if PYTHON_ENV_INITIALIZE.set(()).is_err() { ... }` and returning an error if that call fails
- otherwise, continue with initialization and eventually return `Self` at the end

This will ensure we're guarding against multiple initializations _and_ make the code a bit clearer, I think.